Additional R&D on capbilities of Drive API to plan for next steps

sixside opened the pull request:
Created new client_secrets.json file for local dev (need a way for each of us to keep our own file but ignore push/pull to/from github)
Added ToDo list in comments for next steps

sixside committed:
Additional R&D on capbilities of Drive API to plan for next steps

Created new client_secrets.json file for local dev (need a way for each of us to keep our own file but ignore push/pull to/from github)
Added ToDo list in comments for next steps

sixside commented:
That's hot.

On Sun, Sep 3, 2017 at 6:58 PM, Melissa Quintero Jimison <
notifications@github.com> wrote:

> Merged #1 <https://github.com/melissajimison/travel_gif/pull/1>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/melissajimison/travel_gif/pull/1#event-1232814066>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB_R8L5lHRysssZdk321hB4I40rI0yBEks5se1lUgaJpZM4PLW25>
> .
>

meliquintero commented on quickstart.rb:
```diff
@@ -45,26 +43,68 @@ def authorize
+  #puts aFile.inspect
 end
+
+# note: I don't think exporting of gifs works. :(
```
The export method is to download the file on a different format, Ex: from .doc to PDF.

[Docs here](https://developers.google.com/drive/v3/web/about-sdk#export_and_convert_google_docs)

sixside replied:
yup. boo! 👎

meliquintero commented on quickstart.rb:
```diff
@@ -45,26 +43,68 @@ def authorize
+
+## TODO
+#1 Save a copy of gif file (maybe via cUrl?) to gifs.travel server using webContentLink
+  # Windows PowerShell example => Invoke-WebRequest 'https://drive.google.com/uc?id=0B2tqn9-EHK-ZU3ZOWmc1dHR6OFU&export=download' -OutFile test-ANIMATION.gif
```
There should be a way to display the gifs with the `web_view_link`  directly on `gif.travel` either embedded or adding passing a query param KEY or another sort of authentication to that URL.

sixside replied:
I tried to embed an image in a simple webpage using various URL formats from Google. None of them rendered. I just added sample_embedded_image.htm to master that contains a few of those tests.

That's why I added the to-do item for copying the file to another server -- since I was able to test (successfully) that a non-authenticated user could download the file using the webContentLink URL

If you can figure out a way to embed the image on another site and have the image served directly from Google Drive's servers, that would be better! Save bandwidth and storage! ;)

meliquintero commented on quickstart.rb:
```diff
@@ -45,26 +43,68 @@ def authorize
+  # parse RSS to get pubDate from channel > item[0]
+  # convert pubDate to date
+  # loop through 1000 most recent gifs with Drive.API call {note: that initially need to run through ALL results (not just top 1000)}
+  # if gif.created_time > pubDate then add new item node to rss for each new gif with all metadata
```
I don't understand a word. I would store the gifs data from Google on a data base...

sixside replied:
Do you think a database would be overkill? I'm not sure what advantage would be gained by using a database over using the XML/RSS file. Curious to hear your thoughts.

meliquintero commented on quickstart.rb:
```diff
@@ -45,26 +43,68 @@ def authorize
+
+  # IMPORTANT: Ignore all files if ownedByMe = false
+
+#4 schedule job to run periodically
```
Schedule the job to populate the data base

meliquintero commented on quickstart.rb:
```diff
@@ -45,26 +43,68 @@ def authorize
+
+#4 schedule job to run periodically
+#5 build a web template XMLNS file to display HTML from RSS/XML file. Display 10 gifs per page.
+# future features: user voting of gifs. social share links. UI to sort display based on ratings or most recent
```
👍

sixside commented:
I tried to embed an image in a simple webpage using various URL formats
from Google. None of them rendered. I just added sample_embedded_image.htm
to master.

That's why I added the to-do item for copying the file to another server --
since I was able to test (successfully) that a non-authenticated user could
download the file using the webContentLink URL

If you can figure out a way to embed the image on another site and have the
image served directly from Google Drive's servers, that would be better!
Save bandwidth and storage! ;)

On Mon, Sep 4, 2017 at 2:21 PM, Melissa Quintero Jimison <
notifications@github.com> wrote:

> *@melissajimison* commented on this pull request.
> ------------------------------
>
> In quickstart.rb
> <https://github.com/melissajimison/travel_gif/pull/1#discussion_r136871059>
> :
>
> >  end
> +
> +# note: I don't think exporting of gifs works. :(
> +#exportFile = apiCall.export_file("1hqnXJsVTSgv3H2gsFoyUVKLQTNiaBcOD8Q", "image/gif")
> +
> +
> +
> +## TODO
> +#1 Save a copy of gif file (maybe via cUrl?) to gifs.travel server using webContentLink
> +  # Windows PowerShell example => Invoke-WebRequest 'https://drive.google.com/uc?id=0B2tqn9-EHK-ZU3ZOWmc1dHR6OFU&export=download' -OutFile test-ANIMATION.gif
>
> There should be a way to display the gifs with the web_view_link directly
> on gif.travel either embedded or adding passing a query param KEY or
> another sort of authentication to that URL.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/melissajimison/travel_gif/pull/1#pullrequestreview-60468739>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB_R8MDDALtt0IJN4J_El_HMUkT2KQj6ks5sfGnPgaJpZM4PLW25>
> .
>
